File: leetcode/rust/src/70.climbing-stairs.rs
impl Solution {
    pub fn climb_stairs(n: i32) -> i32 {
        if n == 1{
             return 1;
        }
        if n == 2{
             return 2;
        }

        let mut prev = 1;
        let mut curr = 2;

        for i in 3..(n+1){
            curr += prev;
            prev = curr - prev;
            
        }
        curr
    }
}

File: leetcode/cpp/70.climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n)
    {
        if (n == 1)
            return 1;
        if (n == 2)
            return 2;

        int prev = 1;
        int curr = 2;
        for (int i = 3; i < n + 1; ++i) {
            curr += prev;
            prev = curr - prev;
        }
        return curr;
    }
};
